### PR TITLE
[IMP]lunch: View updated, removed fileds user_id, state and warning m…

### PR DESCRIPTION
**Pull request by ghost**
Task:https://www.odoo.com/web?#id=19877&view_type=form&model=project.task&action=327
Pad: https://pad.odoo.com/p/r.5pEGdTqhd7ylhOZK

### CODE REVIEW

File: addons/lunch/models/lunch.py [hunk 1]
```diff
@@ -39,6 +39,21 @@ class LunchOrder(models.Model):
...
     balance_visible = fields.Boolean(compute='_compute_cash_move_balance', multi='cash_move_balance')
 
+    @api.model
+    def _search_order_state(self, operator, operand):
+        records = []
+        domain = []
+        if operand in ('new', 'confirmed'):
+            if operator == '=':
+                domain = [('order_line_ids.state', '=', 'cancelled')] if operand == 'new' else [('order_line_ids.state', 'in', ('new', 'cancelled'))]
+                not_rec = self.search(domain)
```
Comment (hmo-odoo) [inline addons/lunch/models/lunch.py]: can you make proper domain instead of calling two time search method ?

File: addons/lunch/models/lunch.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ class LunchOrder(models.Model):
                               ('confirmed', 'Received'),
                               ('cancelled', 'Cancelled')],
                              'Status', readonly=True, index=True, copy=False, default='new',
-                             compute='_compute_order_state')
+                             compute='_compute_order_state', store="True")
```
Comment (hmo-odoo) [inline addons/lunch/models/lunch.py]: store should have boolean value...

File: addons/lunch/views/lunch_views.xml [hunk 3]
```diff
@@ -366,35 +370,36 @@
                 <form string='Orders Form' class="o_lunch">
                     <header>
                         <button name="%(action_lunch_order_line_lucky)d" type="action" string="Feeling Lucky" class="oe_highlight"/>
-                        <field name='state' widget='statusbar' statusbar_visible='new,confirmed'/>
+                        <field name='state' widget='statusbar' statusbar_visible='new,confirmed' invisible="context.get('hide_state', False)"/>
                     </header>
                     <sheet>
-                        <field name="balance_visible" invisible="1"/>
+                         <field name="balance_visible" invisible="1"/>
```
Comment (atp-odoo) [inline addons/lunch/views/lunch_views.xml]: unused indentation.

File: addons/lunch/views/lunch_views.xml [hunk 4]
```diff
@@ -366,35 +370,36 @@
...
+                                type="action">
+                                <div class="o_form_field o_stat_info">
+                                    <span class="o_stat_value">
+                                        <field name="cash_move_balance" widget="monetary"/>
+                                    </span>
+                                    <span class="o_stat_text">Balance</span>
+                                </div>
+                            </button>
+                        </div>
+                        <field name="balance_visible" invisible="1"/>
```
Comment (atp-odoo) [inline addons/lunch/views/lunch_views.xml]: two times balance_visible field with invisible ?

**Pull request closed without merging**